### PR TITLE
WIP - Consolidate to vars files and expose all Jenkins Parameters in README

### PR DESCRIPTION
**Pull request by akrzos**
DNM - Do not merge yet until we agree this format is "better/easier" etc.

### CODE REVIEW

File: vars/deploy.yml [hunk 1]
```diff
@@ -25,15 +33,45 @@ introspect_with_retry: "{{ lookup('env', 'OSP_INTROSPECTION_SCRIPT')|default(tru
...
-stack_password: "{{ lookup('env', 'STACK_PASSWORD') }}"
+####################################################################################################
+# Overcloud Variables
+####################################################################################################
+instackenv_json: "{{ lookup('env', 'OSP_INSTACKENV_JSON') }}"
+
+cloud_title: "{{ lookup('env', 'OSP_CLOUD_TITLE') }}"
+
+# Defaults for SLC1 (16 Nodes)
+controller_type: "{{ lookup('env', 'OSP_CONTROLLER_TYPE')|default("1029p", true) }}"
```
Comment (mbruzek) [inline vars/deploy.yml]: You will need to use single quotes around the default value here `'1029p'`

File: vars/deploy.yml [hunk 2]
```diff
@@ -25,15 +33,45 @@ introspect_with_retry: "{{ lookup('env', 'OSP_INTROSPECTION_SCRIPT')|default(tru
...
+cephstorage_type: "{{ lookup('env', 'OSP_CEPHSTORAGE_TYPE')|default("1029p", true) }}"
```
Comment (mbruzek) [inline vars/deploy.yml]: double quote to single quote here too

File: vars/deploy.yml [hunk 3]
```diff
@@ -25,15 +33,45 @@ introspect_with_retry: "{{ lookup('env', 'OSP_INTROSPECTION_SCRIPT')|default(tru
...
+compute_type: "{{ lookup('env', 'OSP_COMPUTE_TYPE')|default("1029p", true) }}"
```
Comment (mbruzek) [inline vars/deploy.yml]: one more !

File: vars/data.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+# Vars that do not require much (if any) adjustment
+
+overcloud_image_password: debug
```
Comment (mbruzek) [inline vars/data.yaml]: Do we want/need to have this password checked in? Can we read this from an environment variable as well?

**Comment by mbruzek** [comment]
I would like to advocate for as few variables as possible to reduce the barrier of entry for someone setting this up. We could use some of the variables to compute other variables. Such as the `cloud_title` and forman url can compute the instackenv.json url. From that we could get the ticket number in ansible. That would reduce the possibility for mistakes.

File: vars/deploy.yml (hunk 1 above)
Comment (akrzos) [reply vars/deploy.yml]: Thanks, I fixed the quote issues now.

File: vars/data.yaml (hunk 4 above)
Comment (akrzos) [reply vars/data.yaml]: This password is just for debugging and requires console access to even use. I do not think there is an issue with keeping it in any of these files, however we can always expose it if others do not feel comfortable with it here.

**Comment by akrzos** [comment]
> I would like to advocate for as few variables as possible to reduce the barrier of entry for someone setting this up. We could use some of the variables to compute other variables. Such as the cloud_title and forman url can compute the instackenv.json url. From that we could get the ticket number with just Ansible code. Since these variables are all tied together that would reduce the possibility for mistakes.

I agree we could hone this down a bit.  Do we want to save that idea for the next p/r?

**Comment by mbruzek** [comment]
I would really like to eliminate the variable in this change. Since it was already there I created a PR to  remove it https://github.com/redhat-performance/tripleo-quickstart-scalelab/issues/99 in a future PR

**Pull request merged**